tools/encryption/encryptionutils/encryption-helper.go

```diff
@@ -0,0 +1,243 @@
+// Helper functions for the SwG Encryption Script.
+package encryptionutils
+
+import (
+	"bytes"
+    "encoding/base64"
+	"fmt"
+	"io"
+	"golang.org/x/net/html"
+	"golang.org/x/net/html/atom"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/tink/go/aead"
+	"github.com/google/tink/go/hybrid"
+	"github.com/google/tink/go/insecurecleartextkeyset"
+	"github.com/google/tink/go/keyset"
+	"github.com/google/tink/go/core/registry"
+	gcmpb "github.com/google/tink/proto/aes_gcm_go_proto"
+	tinkpb "github.com/google/tink/proto/tink_go_proto"
+	"net/http"
+	"strings"
+)
+
+const AES_GCM_KEY_URL string = "type.googleapis.com/google.crypto.tink.AesGcmKey" 
+const AES_GCM_KEY_SIZE uint32 = 16
+
+// Public function to generate an encrypted HTML document given the original.
+func GenerateEncryptedDocument(html_str string, public_key_url string, access_requirement string) (string, error) {
+	keyManager, err := registry.GetKeyManager(AES_GCM_KEY_URL)
+	if err != nil {
+		return "", err
+	}
+	key, err := generateNewAesGcmKey(keyManager)
+	if err != nil {
+		return "", err
+	}
+	ks := createAesGcmKeyset(key)
+	r := strings.NewReader(html_str)
+	parsed_html, err := html.Parse(r)
+	if err != nil {
+		return "", err
+	}
+	encrypted_sections := getAllEncryptedSections(parsed_html)
```

Comment:
Move all parsing and encryption to as late as possible. IOW, set up your required dependencies first (key, keyset, publicKey).

Reply:
Moved key handling up in the flow.